### ISSUE TITLE
Reproduced Performance Discussion

### ISSUE THREAD

**Issue by PkuRainBow** [junfu1115/DANet#6]
Very interesting! 

I find that you employ three independent loss functions on the predictions of the "PAM branch" and "CAM branch".

I am wondering the influence of the loss functions on these two branches and it would be great if you could share the related experiments.

Besides, I want to share my reproduced performance with your PAM and CAM based on the [OCNet](https://github.com/PkuRainBow/OCNet), (especially, we only employ one loss function and employ lr=0.01 and wd=0.0005 following the default settings within OCNet)

we report the mIoU  on training set /validation set here: **82.0/74.4** (single scale).

Could anyone help us to check where is the problem?

Here we also provide the network configuration,

```
class ResNet(nn.Module):
    def __init__(self, block, layers, num_classes):
        self.inplanes = 128
        super(ResNet, self).__init__()
        self.conv1 = conv3x3(3, 64, stride=2)
        self.bn1 = BatchNorm2d(64)
        self.relu1 = nn.ReLU(inplace=False)
        self.conv2 = conv3x3(64, 64)
        self.bn2 = BatchNorm2d(64)
        self.relu2 = nn.ReLU(inplace=False)
        self.conv3 = conv3x3(64, 128)
        self.bn3 = BatchNorm2d(128)
        self.relu3 = nn.ReLU(inplace=False)
        self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
        self.relu = nn.ReLU(inplace=False)
        self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1, ceil_mode=True) # change
        self.layer1 = self._make_layer(block, 64, layers[0])
        self.layer2 = self._make_layer(block, 128, layers[1], stride=2)
        self.layer3 = self._make_layer(block, 256, layers[2], stride=1, dilation=2)
        self.layer4 = self._make_layer(block, 512, layers[3], stride=1, dilation=4, multi_grid=(1,1,1))

        # extra added layers
        self.head = DANetHead(2048, num_classes)
        self.dsn = nn.Sequential(
            nn.Conv2d(1024, 512, kernel_size=3, stride=1, padding=1),
            InPlaceABNSync(512),
            nn.Dropout2d(0.1),
            nn.Conv2d(512, num_classes, kernel_size=1, stride=1, padding=0, bias=True)
            )
```

Here we provide the configuration of the DAHead,

```
class DANetHead(nn.Module):
    def __init__(self, in_channels, out_channels, norm_layer=InPlaceABNSync):
        super(DANetHead, self).__init__()
        inter_channels = in_channels // 4
        self.conv5a = nn.Sequential(nn.Conv2d(in_channels, inter_channels, 3, padding=1, bias=False),
                                   norm_layer(inter_channels),
                                   )    
        self.conv5c = nn.Sequential(nn.Conv2d(in_channels, inter_channels, 3, padding=1, bias=False),
                                   norm_layer(inter_channels),
                                   )
        self.sa = PAM_Module(inter_channels)
        self.sc = CAM_Module(inter_channels)
        self.conv51 = nn.Sequential(nn.Conv2d(inter_channels, inter_channels, 3, padding=1, bias=False),
                                   norm_layer(inter_channels),
                                   )
        self.conv52 = nn.Sequential(nn.Conv2d(inter_channels, inter_channels, 3, padding=1, bias=False),
                                   norm_layer(inter_channels),
                                   )
        self.conv8 = nn.Sequential(nn.Dropout2d(0.1, False), nn.Conv2d(512, out_channels, 1))
```

 I guess the main difference is only about the choice of the SyncBN, we employ the  InPlaceABNSync (contains PReLU) following OCNet.

**Comment by junfu1115** [comment]
Thank you for paying attention to our work
Maybe the BN or some other settings in OCNet are different from ours, we adopt the Synchronized BN from [PyTorch-Encoding](https://github.com/zhanghang1989/PyTorch-Encoding), which is a good code framework.
We have released our training settings and the code help you reproduced our performance.

**Comment by zimenglan-sysu-512** [comment]
@PkuRainBow 
will you release the code of reproducing the performance of  82.0/74.4 (single scale) in your OCNet repo?

**Comment by XiaLiPKU** [comment]
> 
> 
> Thank you for paying attention to our work
> Maybe the BN or some other settings in OCNet are different from ours, we adopt the Synchronized BN from [PyTorch-Encoding](https://github.com/zhanghang1989/PyTorch-Encoding), which is a good code framework.
> We have released our training settings and the code help you reproduced our performance.

I have the same problem as @PkuRainBow. 
Detailly, I adopt the implementation of Sync BN from @Vacancy and @tamakoji.
But both of them don't work on CAM.
